src/jobflow/core/flow.py

```diff
@@ -647,14 +657,33 @@ def update_metadata(
 
         >>> flow.update_metadata({"tag": "addition_job"})
         """
-        for job in self:
-            job.update_metadata(
-                update,
-                name_filter=name_filter,
-                function_filter=function_filter,
-                dict_mod=dict_mod,
-                dynamic=dynamic,
-            )
+        from jobflow.utils.dict_mods import apply_mod
+
+        if target in ["flow", "both"] and name_filter in (None, self.name):
+            if dict_mod:
+                apply_mod(update, self.metadata)
+            else:
+                self.metadata.update(update)
+
+        if target in ["jobs", "both"]:
+            for job in self:
```

Comment:
Here `job` could actually be a nested flow. So you should still iterate over these if `target = "flow"`.

Reply:
perhaps the naming could be improved. `target = "flow"` isn't meant to update the metadata of all nested flows (to the exclusion of jobs) but rather only the flow itself, not any of its jobs or nested flows.

how about we rename from `target: Literal["flow", "jobs", "both"] = "both"` to `target: Literal["self", "nested", "both"] = "both"`? unless you think it's important to have more control, i.e. be able to only update nested `Flow` or `Job` metadata. there might be use cases for that. in which case maybe we want `target: Literal["self", "nested", "jobs", "flows", "both"] = "both"`?

Reply:
to clarify this could be the doc string:

```
        target
            Specifies where to apply the metadata update. Options are:

            - "self": Update only the Flow's own metadata
            - "nested": Update only the metadata of Jobs/Flows within the Flow
            - "jobs": Update only the metadata of Jobs within the Flow
            - "flows": Update only the metadata of Flows within the Flow
            - "all": Update both the Flow's metadata and nested Job+Flow metadata (default)
```

Reply:
I think this is overcomplicating it. If you want to select specific flows or jobs then you can pass in a name or class filter. So no need for the extra options. The main thing is the API should be consistent between jobs/flows.

Reply:
makes sense. then the easiest thing would be to get rid of `target` altogether and use the `"all"` behavior?

Reply:
> I think the benefit of target is that it would be a shortcut for `class=Flow` vs `class=Job`. I would be happy either having the target option or specifying the shortcut in the docstring.

i don't follow. by `class=Job|Flow`, did you mean `filter_function=Job|Flow`? the job variant `filter_function=Job` doesn't seem to work the way you're suggesting so maybe you meant sth else?

Reply:
Yes, I was thinking of `class_filter`. I think the `callback_filter` you proposed sounds very flexible and would be useful!